artifactory.py

```diff
@@ -1514,6 +1519,10 @@ def __new__(cls, *args, **kwargs):
         only then add auth information.
         """
         obj = pathlib.Path.__new__(cls, *args, **kwargs)
+     
+        if sys.version_info.major == 3 and sys.version_info.minor >= 12:
```

Comment:
We should not have version specific behaviour if this can be avoided. If positional arguments can be passed in all supported earlier versions, then we should be doing it for all versions.

Why is `__init__` only being called here for 3.12+?